src/vmm/src/vstate/vcpu/mod.rs

```diff
@@ -450,15 +450,21 @@ impl Vcpu {
             Ok(run) => match run {
                 VcpuExit::MmioRead(addr, data) => {
                     if let Some(mmio_bus) = &self.kvm_vcpu.mmio_bus {
+                        let start_time =
+                            utils::time::get_time_us(utils::time::ClockType::Monotonic);
```

Comment:
Maybe we can move this (and `record_latency_summary` line) into `LatencyAggregateMetricWriter`(naming is hard) so that it gets the current time on creation and writes to the metric on drop?

Reply:
Thanks for the suggestion, updated the code accordingly